qt/widgets/common/src/FitScriptGeneratorModel.cpp

```diff
@@ -5,148 +5,99 @@
 #include <iterator>
 #include <stdexcept>
 
 using namespace Mantid::API;
+using namespace MantidQt::MantidWidgets;
 
 namespace {
+Mantid::Kernel::Logger g_log("FitScriptGeneratorModel");
 
 IFunction_sptr createIFunction(std::string const &functionString) {
   return FunctionFactory::Instance().createInitialized(functionString);
 }
 
-CompositeFunction_sptr toComposite(IFunction_sptr function) {
-  return std::dynamic_pointer_cast<CompositeFunction>(function);
+std::string removeTopFunctionIndex(std::string const &functionIndex) {
+  auto resultPrefix = functionIndex;
+  auto const firstDotIndex = resultPrefix.find(".");
+  if (firstDotIndex != std::string::npos)
+    resultPrefix.erase(0, firstDotIndex + 1);
+  return resultPrefix;
 }
 
-CompositeFunction_sptr createEmptyComposite() {
-  return toComposite(createIFunction("name=CompositeFunction"));
+std::string replaceTopFunctionIndexWith(std::string const &functionIndex,
+                                        std::size_t const &newIndex) {
+  return "f" + std::to_string(newIndex) + "." +
+         removeTopFunctionIndex(functionIndex);
 }
 
-MultiDomainFunction_sptr
-createMultiDomainFunction(std::size_t const &numberOfDomains) {
-  return FunctionFactory::Instance().createInitializedMultiDomainFunction(
-      "name=CompositeFunction", numberOfDomains);
+std::string getTieRHS(std::string const &tie) {
+  auto const tieSplit = splitStringBy(tie, "=");
+  return tieSplit.size() > 1 ? tieSplit[1] : tieSplit[0];
 }
 
-std::vector<std::string> splitStringBy(std::string const &str,
-                                       std::string const &delimiter) {
-  std::vector<std::string> subStrings;
-  boost::split(subStrings, str, boost::is_any_of(delimiter));
-  subStrings.erase(std::remove_if(subStrings.begin(), subStrings.end(),
-                                  [](std::string const &subString) {
-                                    return subString.empty();
-                                  }),
-                   subStrings.cend());
-  return subStrings;
+std::string getLocalTie(std::string const &tie,
+                        FittingMode const &fittingMode) {
+  if (!tie.empty() && fittingMode == FittingMode::SIMULTANEOUS)
+    return removeTopFunctionIndex(getTieRHS(tie));
+  return tie;
 }
 
-bool isSinglePrefix(std::string const &prefix) {
-  auto const subStrings = splitStringBy(prefix, "f.");
-  return subStrings.size() == 1;
+bool isFunctionIndex(std::string const &str) {
+  auto const subStrings = splitStringBy(str, "f");
+  if (subStrings.size() == 1)
+    return isNumber(subStrings[0]);
+  return false;
 }
 
-std::size_t getPrefixIndexAt(std::string const &prefix,
-                             std::size_t const &index) {
-  auto const subStrings = splitStringBy(prefix, "f.");
-  return std::stoull(subStrings[index]);
+bool isSameDomain(std::size_t const &domainIndex,
+                  std::string const &parameter) {
+  if (!parameter.empty() && !isNumber(parameter))
```

Comment:
what happens if we have the same parameter in multiple functions?

Reply:
This should be ok as it is only checking that the domain index in the parameter is equal to the domain index provided. i.e. it doesn't check the actual parameter, just the first number at the start of the parameter. (which would be 0 in `f0.f1.Height`)

Reply:
so `parameter` is the full name `f0.f1.Height`?

Reply:
Yes `parameter` is the full parameter. I will rename this to make it clearer.